internal/command/project.go

```diff
@@ -165,16 +165,34 @@ func (c *Commands) getProjectByID(ctx context.Context, projectID, resourceOwner
 	return projectWriteModelToProject(projectWriteModel), nil
 }
 
+func (c *Commands) projectAggregateByID(ctx context.Context, projectID, resourceOwner string) (*eventstore.Aggregate, domain.ProjectState, error) {
+	result, err := c.eventstore.Search(
+		ctx,
+		map[eventstore.SearchFieldType]any{
+			eventstore.SearchFieldTypeObjectType:     project.ProjectSearchType,
+			eventstore.SearchFieldTypeObjectID:       projectID,
+			eventstore.SearchFieldTypeObjectRevision: project.ProjectObjectRevision,
+			eventstore.SearchFieldTypeFieldName:      project.ProjectStateSearchField,
+			eventstore.SearchFieldTypeResourceOwner:  resourceOwner,
+		},
+	)
+	if err != nil || len(result) == 0 {
+		return nil, domain.ProjectStateUnspecified, zerrors.ThrowNotFound(err, "COMMA-NDQoF", "Errors.Project.NotFound")
+	}
+	state, err := eventstore.NumericResultValue[domain.ProjectState](result[0])
```

Comment:
Isn't storing a specific state outside the scope of the search table. Now it feels it will start to act as a projection. I thought the issue we tried to solve is payload queries. Once we resolved the aggregate ID, any state can be obtained directly from the eventstore, right?

I'm just raising this, as not it is just a numeric state, but before you know it we can justify all sorts of stuff in the search table and it becomes a bloated thing.